### ISSUE TITLE
MethodCallError hides real errors

### ISSUE THREAD

**Issue by lindycoder** [flexmock/flexmock#21]
Hello!

I found a little inconvenience with the feedback i get when using expectations when using py.test:

If an expectation is not called, the hook raises a MethodCallError even if the test itself failed for an error.  Such an error can even be a MethodSignatureError, which means that if my params are wrong and my expectation has a .once() i don't see the MethodSignatureError, i see a MethodCallError saying i haven't called it.   This is kind of hard to deal with as you need to enter debug mode, break on the line that is generating the error and follow the path up to see what was the error before the teardown launches and the hook fires the MethodCallError.

I created a simple example here : https://github.com/lindycoder/flexmock-example

Where i create an expectation then raise an exception, i would expect to see the exception it self now all i see is the MethodCallError.

I don't know if that can be fixed at all, it would if py.test lets you know the state of the test, on which you could decide whether to throw the MethodCallError only if the test "is about the pass".

So this is not major, just a little inconvenience.

Thanks you for your excellent work!

**Comment by mattduck** [comment]
Hey @lindycoder, I've run into this too. Just had a very quick look and the original exception is being overridden here: https://github.com/bkabrda/flexmock/commit/a10b5412ad2bd4bfdc177070c652552fbc33287c#diff-2979e8cf7da2d8484d298236311f1cb8R1213

```
if ret.excinfo is not None and teardown.excinfo is None:
    teardown.excinfo = ret.excinfo
```

The original `ret.excinfo` is overridden any time flexmock raises. I might submit a fix at the weekend. You can temporarily patch with:

```
# if ret.excinfo is not None and teardown.excinfo is None:
#     teardown.excinfo = ret.excinfo
if ret.excinfo:
    teardown.excinfo = ret.excinfo
```

**Comment by bkabrda** [comment]
Fixed in #22. Closing this issue.
